src/models/syncRule.ts

```diff
@@ -133,11 +149,146 @@ export default class SyncRule {
     }
 
     public updateMetadataIds(metadataIds: string[]): SyncRule {
+        if (_.isEqual(this.metadataIds, metadataIds)) {
+            return SyncRule.build({
+                ...this.syncRule,
+                builder: {
+                    ...this.syncRule.builder,
+                    metadataIds,
+                },
+            });
+        } else {
+            // When metadataIds really has changed we should reset
+            // useDefaultIncludeExclude and metadataIncludeExcludeRules
+            return SyncRule.build({
+                ...this.syncRule,
+                builder: {
+                    ...this.syncRule.builder,
+                    metadataIds,
+                    useDefaultIncludeExclude: true,
+                    metadataIncludeExcludeRules: undefined,
+                },
+            });
+        }
```

Comment:
Could we assume that if ``updateMetadataIds`` has been called, the metadataIds did change?

If we are receiving false-positives that's a bug and we shouldn't tackle with an ``isEqual`` on model.

I would particularly do the following without any other check:

```ts
            return SyncRule.build({
                ...this.syncRule,
                builder: {
                    ...this.syncRule.builder,
                    metadataIds,
                    useDefaultIncludeExclude: true,
                    metadataIncludeExcludeRules: {},
                },
            });
```

Reply:
No, we can not assume when updateMetadataIds has been called, the metadataIds did change. Currently, this method is invoked when the user to navigate for steps without making any changes. So this logic must be controlled in the model for several reasons: protect its invariants (model cannot reach an inconsistent state), code reuse, testing etc..